### ISSUE TITLE
Lower bound for GasLimit is hardcoded and can not be changed by parameters

### ISSUE THREAD

**Issue by hesusruiz** [ConsenSys/quorum#1434]
#### System information

Geth version: `any after Apr 16, 2019`
OS & Version: Linux

#### Expected behaviour
Using `--miner.gastarget` and `--miner.gaslimit` in the command line of Validators would set the lower and upper bounds for the gas limit in the network, so the validators would:
1. Reject transactions specifying gas above `--miner.gaslimit`
2. Never create blocks using gas above `--miner.gaslimit`

#### Actual behaviour
In function `CalcGasLimit` in file `core/block_validator.go` the minimum absolute value of the next block gas limit is hardcoded by `params.MinGasLimit`, which is a constant with value `700000000` and can not be reduced using configuration values.

The minimum actual gas limit in blocks is `parent.GasLimit() - decay`

#### Steps to reproduce the behaviour
- Set `--miner.gastarget` and `--miner.gaslimit` to values lower than `700000000` (eg. 8000000 and 15000000).
- Send transactions with gas greater than `--miner.gaslimit` and lower than `700000000`
- They are accepted

#### Additional info

```go
limit := parent.GasLimit() - decay + contrib
if limit < params.MinGasLimit {
	limit = params.MinGasLimit
}
// If we're outside our allowed gas range, we try to hone towards them
if limit < gasFloor {
	limit = parent.GasLimit() + decay
	if limit > gasFloor {
		limit = gasFloor
	}
} else if limit > gasCeil {
	limit = parent.GasLimit() - decay
	if limit < gasCeil {
		limit = gasCeil
	}
}
```

From the above extract of function `CalcGasLimit`, if `gasFloor` and `gasCeil` are below the constant `params.MinGasLimit`, the new `limit` will be `limit = parent.GasLimit() - decay`, even if all the previous blocks are almost empty.

**Comment by baptiste-b-pegasys** [comment]
Hello

Currently we are based on geth 1.10.2, we modify the geth's code base sparsely for go-quorum.

https://github.com/ethereum/go-ethereum/blob/97d11b0187b4695ccf44e3b71b54155fe405a36f/core/block_validator.go#L109

Is this issue has to be opened in the go-ethereum side too?

**Comment by hesusruiz** [comment]
Thanks for the reply.
The algorithms for calculating the next block gas limit are exactly the same in Quorum and Ethereum.

The problem is that the constant `params.MinGasLimit` is just 5.000 in Ethereum and 70.000.000 in Quorum
https://github.com/ConsenSys/quorum/blob/d258a138864999157cbcd7be4037b70d7cb772e5/params/protocol_params.go#L28

https://github.com/ethereum/go-ethereum/blob/b3b8b268eb585dfd3c1c9e9bbebc55968f3bec4b/params/protocol_params.go#L23

That means that in Ethereum we can set `miner.gaslimit` and `miner.gastarget`to anything reasonable for our network.
But this is not the case in Quorum, where we are forced to set those values to something bigger than 70M, which is way too much for our requirements.

At the begining, the MinGasLimit in Quorum was 8M, and we were running fine. But when we upgraded the clients, we did not notice it until we got roge transactions with big gas values.

If the parameter `MinGasLimit` can not be reverted back in Quorum, a solution for us would be to be able to set it via a configuration parameter.
This is the reason for the word "hardcoded" in the incident: it is also hardcoded in Ethereum, but because in Quorum it is so big, we can not set our values for gaslimit and gastarget.
